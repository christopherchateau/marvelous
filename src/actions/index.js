export const storeCharacter = (character) => ({
  type: 'STORE_CHARACTER',
  character
})

export const updateStorageDetails = (currentIndex, length) => ({
  });
