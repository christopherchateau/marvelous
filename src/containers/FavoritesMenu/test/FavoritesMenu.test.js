import React from "react";
import { shallow } from "enzyme";
import { showFavorites, updateStorageDetails } from "../../../actions";
import { FavoritesMenu, mapDispatchToProps, mapStateToProps } from "../index";

describe("FavoritesMenu", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <FavoritesMenu
        storedCharacters={[]}
        favoriteCharacters={[]}
        dispatchShowFavorites={jest.fn()}
        dispatchStorageDetailsUpdate={jest.fn()}
      />
    );
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("mapDispatchToProps", () => {
    it("should dispatch showFavorites when dispatchShowFavorites is called", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = showFavorites(true);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.dispatchShowFavorites(true);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it("should dispatch updateStorageDetails when dispatchShowFavorites is called", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = updateStorageDetails({
        currentIdex: 1,
        count: 3
      });
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.dispatchStorageDetailsUpdate({ currentIdex: 1, count: 3 });
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

  describe("mapStateToProps", () => {
    it("should return an array of characters", () => {
      const mockState = {
        characters: [{ name: "Spider-Man", id: 1, favorited: false }]
      };
      const expected = [{ name: "Spider-Man", id: 1, favorited: false }];
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps.storedCharacters).toEqual(expected);
    });

    it("should return array of favorite characters", () => {
      const mockState = {
        characters: [
          { name: "Spider-Man", id: 1, favorited: false },
          { name: "Wolverine", id: 2, favorited: true }
        ]
      };
      const expected = [{ name: "Wolverine", id: 2, favorited: true }];
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps.favoriteCharacters).toEqual(expected);
    });
  });
});
