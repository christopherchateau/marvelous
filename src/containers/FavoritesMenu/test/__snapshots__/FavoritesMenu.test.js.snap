// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesMenu should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoritesMenu
    dispatchShowFavorites={[MockFunction]}
    dispatchStorageDetailsUpdate={[MockFunction]}
    favoriteCharacters={Array []}
    storedCharacters={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="fav-menu empty"
      >
        <img
          alt="Captain America"
          className="capt-america"
          src="capt-america.gif"
        />
        <h3>
          No favorites selected
        </h3>
      </section>,
      "className": "FavoritesMenu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Captain America"
            className="capt-america"
            src="capt-america.gif"
          />,
          <h3>
            No favorites selected
          </h3>,
        ],
        "className": "fav-menu empty",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Captain America",
            "className": "capt-america",
            "src": "capt-america.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No favorites selected",
          },
          "ref": null,
          "rendered": "No favorites selected",
          "type": "h3",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="fav-menu empty"
        >
          <img
            alt="Captain America"
            className="capt-america"
            src="capt-america.gif"
          />
          <h3>
            No favorites selected
          </h3>
        </section>,
        "className": "FavoritesMenu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Captain America"
              className="capt-america"
              src="capt-america.gif"
            />,
            <h3>
              No favorites selected
            </h3>,
          ],
          "className": "fav-menu empty",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Captain America",
              "className": "capt-america",
              "src": "capt-america.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No favorites selected",
            },
            "ref": null,
            "rendered": "No favorites selected",
            "type": "h3",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
